packages/agreement-process/test/testUpdateAgreement.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable functional/no-let */
+import {
+  getMockAgreement,
+  getRandomAuthData,
+  decodeProtobufPayload,
+  randomArrayItem,
+} from "pagopa-interop-commons-test";
+import {
+  AgreementId,
+  AgreementUpdatedV1,
+  agreementState,
+  generateId,
+} from "pagopa-interop-models";
+import { describe, expect, it } from "vitest";
+import { v4 as uuidv4 } from "uuid";
+import { toAgreementV1 } from "../src/model/domain/toEvent.js";
+import {
+  agreementNotFound,
+  agreementNotInExpectedState,
+  operationNotAllowed,
+} from "../src/model/domain/errors.js";
+import { agreementUpdatableStates } from "../src/model/domain/validators.js";
+import { addOneAgreement, readLastAgreementEvent } from "./utils.js";
+import {
+  agreementService,
+  agreements,
+  postgresDB,
+} from "./agreementService.integration.test.js";
+
+export const testUpdateAgreement = (): ReturnType<typeof describe> =>
+  describe("update agreement", () => {
+    it("should succeed when requester is Consumer and the Agreement is in an updatable state", async () => {
+      const agreement = {
+        ...getMockAgreement(),
+        state: randomArrayItem(agreementUpdatableStates),
+      };
+      await addOneAgreement(agreement, postgresDB, agreements);
+      const authData = getRandomAuthData(agreement.consumerId);
+      await agreementService.updateAgreement(
+        agreement.id,
+        { consumerNotes: "Updated consumer notes" },
+        authData,
+        uuidv4()
+      );
+
+      const agreementEvent = await readLastAgreementEvent(
+        agreement.id,
+        postgresDB
+      );
+
+      expect(agreementEvent).toMatchObject({
+        type: "AgreementUpdated",
+        event_version: 1,
+        version: "0",
```

Comment:
🤔  just a minor doubt, in this case `version` should be equal to **"1"** ? 
 I expect this event like an update of early created agreement with same ID, so in the store event should already exists the `stream_id == agreement.id` and `version == 0` , for event the event type `AgreementAdded`.
By the way I suppose we have some error in `writeAgreementInEventstore` implementation,  why this methods are using the "catalog" schema ?
[See here](https://github.com/pagopa/interop-be-monorepo/blob/ca8cb74656211d56baa8bad6e543ca89cb83e4f2/packages/agreement-process/test/utils.ts#L42)

Reply:
You're right - and the issue was the wrong schema string that in some previous test refactors we forgot to change to `"agreement"`.

In the last commit, I improved the way the test event store functions are typed, to avoid these kinds of errors in the future, and fixed the issue you pointed out, and indeed the test now fails if I put version `0` and not `1`:

<img width="461" alt="image" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/b90c89ba-e5b5-4c71-ac4b-2c0dd2415095">

Reply:
@Viktor-K CI tests are failing because I need to port the fix to the other services outside catalog and agreement - I'm gonna do it and then put the PR back in review